packages/event-consumer/src/index.ts

```diff
@@ -17,16 +17,18 @@ const kafka = new Kafka({
 const consumer = kafka.consumer({ groupId: "my-group" });
 await consumer.connect();
 
-// process.on("SIGINT", function () {
-//   consumer.disconnect().then(
-//     () => console.log("Disconnected"),
-//     (err) => console.log(err)
-//   );
-// });
+function exitGracefully(): void {
+  consumer.disconnect().finally(() => {
+    logger.info("Consumer disconnected");
+    process.exit(0);
+  });
+}
+
+process.on("SIGINT", exitGracefully);
+process.on("SIGTERM", exitGracefully);
```

Comment:
Node doesn't support asynchronous signal handlers. As a result, there might be no guarantee that the disconnection always occurs. There's also something strange about this code: the process exists with `ELIFECYCLE  Command failed with exit code 1.` (maybe because the promise is unhandled?).

But, in practice and considering that signal handling is needed for local developer usage and not in production, this looks OK to me. The issue was that, after stopping and restarting the consumer, the next connection attempt was slow because it waited for a timeout since the connection was not closed properly (https://github.com/tulios/kafkajs/issues/1485). Now it's much faster, so the disconnection is happening.